tracing-core/src/collect.rs

```diff
@@ -436,14 +437,14 @@ pub trait Collect: 'static {
     /// # Safety
     ///
     /// The [`downcast_ref`] method expects that the pointer returned by
-    /// `downcast_raw` is non-null and points to a valid instance of the type
+    /// `downcast_raw` points to a valid instance of the type
     /// with the provided `TypeId`. Failure to ensure this will result in
     /// undefined behaviour, so implementing `downcast_raw` is unsafe.
     ///
     /// [`downcast_ref`]: #method.downcast_ref
```

Comment:
...and if we make `NonNull` into a link above, we should add
```suggestion
    /// [`downcast_ref`]: #method.downcast_ref
    /// [`NonNull`]: core::ptr::NonNull
```